src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.cpp

```diff
@@ -78,17 +172,63 @@ mortars_apply_impl(const Element<Dim>& element,
       normal_covector_quantities{};
   for (const auto& [direction, neighbors] : element.neighbors()) {
     normal_covector_quantities[direction] = std::nullopt;
-    for (const auto& neighbor : neighbors) {
-      const DirectionalId<Dim> mortar_id{direction, neighbor};
-      mortar_meshes.emplace(
-          mortar_id,
-          ::dg::mortar_mesh(
-              volume_mesh.slice_away(direction.dimension()),
-              neighbor_mesh.at(mortar_id).slice_away(direction.dimension())));
-      // Since no communication needs to happen for boundary conditions
-      // the temporal id is not advanced on the boundary, so we only need to
-      // initialize it on internal boundaries
-      mortar_next_temporal_ids.insert({mortar_id, next_temporal_id});
+    const Mesh<Dim - 1> face_mesh =
+        volume_mesh.slice_away(direction.dimension());
+    const domain::FaceType face_type = element.face_types().at(direction);
+    switch (face_type) {
+      // NOLINTNEXTLINE(bugprone-branch-clone)
+      case (domain::FaceType::Uninitialized):
+        [[fallthrough]];
+      case (domain::FaceType::External):
+        [[fallthrough]];
+      case (domain::FaceType::Topological): {
+        const std::string message =
+            MakeString{} << "Cannot have FaceType " << face_type
+                         << " in direction " << direction
+                         << " in which there are neighboring elements.";
+        ERROR(message);
+      }
```

Comment:
Is there a way to test the error cases? If so, could you please add the test(s)?

Reply:
I can't think of a way as I don't think currently I can create an Element that violates this.